### ISSUE TITLE
Analyzer hangs when analyzing some files

### ISSUE THREAD

**Issue by DartBot** [dart-lang/sdk#23043]
*This issue was originally filed by @astashov*
_____
**What steps will reproduce the problem?**
1. Clone the repo https://github.com/astashov/analyzer-hanging
2. pub get
3. dart main.dart /path/to/dart-sdk /path/to/packages 

**What is the expected output? What do you see instead?**

I'm expecting to get a resolved CompilationUnit. But it just hangs forever while trying to resolve it on the last line. 

**What version of the product are you using?**
Dart 1.9.1

**On what operating system?**
MacOSX 10.9.5

**What browser (if applicable)?**
Server-side

**Please provide any additional information below.**

**Comment by dgrove** [comment]
@bwilkerson - is this still relevant?

**Comment by mit-mit** [comment]
@bwilkerson, this is looking a little stale. Is this still being worked on?

**Comment by danrubel** [comment]
Yes, this will be addressed as part of landing the new task model.

**Comment by bwilkerson** [comment]
The new task model has now been enabled in the bleeding edge builds and should be in the next dev build.

**Comment by astashov** [comment]
@bwilkerson can I find any details somewhere about that new task model (like, what this is)?

**Comment by bwilkerson** [comment]
The new task model is a different internal implementation of the tasks / steps / phases in the analyzer. The old implementation was more coarse-grained. In order to analyze a single library L, we ended up having to analyze all libraries that L was circularly dependent on as a single large task. This ended up conflicting with some memory management support we had added, causing the problem you were seeing. The new task model is much finer-grained, which, in addition to solving the problem you ran into, also means that we don't have to do nearly as much work before we can get analysis results for a single library (which should improve the performance of tooling such as IntelliJ or Atom).

Because it's an implementation detail the only documentation currently available is in documentation comments in the code.
